indico/modules/events/surveys/templates/display/survey_list.html

```diff
@@ -30,24 +30,29 @@
                         {% if was_survey_submitted(survey) %}
                             <a class="i-button highlight disabled"
                                title="{% trans %}Survey already filled out{% endtrans %}">
-                                {% trans %}Fill out{% endtrans %}
+                                {%- trans %}Fill out{% endtrans -%}
                             </a>
                         {% elif not survey.is_active %}
                             <a class="i-button highlight disabled"
                                title="{% trans %}Survey is closed{% endtrans %}">
-                                {% trans %}Fill out{% endtrans %}
+                                {%- trans %}Fill out{% endtrans -%}
+                            </a>
+                        {% elif is_submission_in_progress(survey) %}
```

Comment:
This results in `O(n)` queries. Not a big deal here since you rarely have tons of surveys and it's not on the main page of an event but it'd be nicer to get rid of this util and instead query the list of surveys within the event for which the user has a pending submission and put them in a set - then you can just check here whether the survey is in that set.